config/src/test/kotlin/org/springframework/security/config/web/server/ServerCsrfDslTests.kt

```diff
@@ -161,70 +160,72 @@ class ServerCsrfDslTests {
-        companion object {
-            var ACCESS_DENIED_HANDLER: ServerAccessDeniedHandler = mock(ServerAccessDeniedHandler::class.java)
-        }
+
+        val serverAccessDeniedHandler: ServerAccessDeniedHandler = mockk()
 
         @Bean
         open fun springWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
             return http {
                 csrf {
-                    accessDeniedHandler = ACCESS_DENIED_HANDLER
+                    accessDeniedHandler = serverAccessDeniedHandler
                 }
             }
         }
     }
 
     @Test
     fun `csrf when custom token repository then repository used`() {
-        `when`(CustomCsrfTokenRepositoryConfig.TOKEN_REPOSITORY.loadToken(any()))
-                .thenReturn(Mono.just(this.token))
         this.spring.register(CustomCsrfTokenRepositoryConfig::class.java).autowire()
+        val customCsrfTokenRepositoryConfig = this.spring.context.getBean(CustomCsrfTokenRepositoryConfig::class.java)
+        every {
+            customCsrfTokenRepositoryConfig.serverCsrfTokenRepository.loadToken(any())
+        } returns Mono.just(this.token)
 
         this.client.post()
                 .uri("/")
                 .exchange()
 
-        Mockito.verify(CustomCsrfTokenRepositoryConfig.TOKEN_REPOSITORY)
-                .loadToken(any())
+        verify(exactly = 1) { customCsrfTokenRepositoryConfig.serverCsrfTokenRepository.loadToken(any()) }
     }
 
     @EnableWebFluxSecurity
     @EnableWebFlux
     open class CustomCsrfTokenRepositoryConfig {
-        companion object {
-            var TOKEN_REPOSITORY: ServerCsrfTokenRepository = mock(ServerCsrfTokenRepository::class.java)
-        }
+
+        val serverCsrfTokenRepository: ServerCsrfTokenRepository = mockk()
 
         @Bean
         open fun springWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
             return http {
                 csrf {
-                    csrfTokenRepository = TOKEN_REPOSITORY
+                    csrfTokenRepository = serverCsrfTokenRepository
                 }
             }
         }
     }
 
     @Test
     fun `csrf when multipart form data and not enabled then denied`() {
-        `when`(MultipartFormDataNotEnabledConfig.TOKEN_REPOSITORY.loadToken(any()))
-                .thenReturn(Mono.just(this.token))
-        `when`(MultipartFormDataNotEnabledConfig.TOKEN_REPOSITORY.generateToken(any()))
-                .thenReturn(Mono.just(this.token))
         this.spring.register(MultipartFormDataNotEnabledConfig::class.java).autowire()
+        val multipartFormDataNotEnabledConfig = this.spring.context.getBean(MultipartFormDataNotEnabledConfig::class.java)
+        every {
+            multipartFormDataNotEnabledConfig.serverCsrfTokenRepository.loadToken(any())
+        } returns Mono.just(this.token)
+        every {
+            multipartFormDataNotEnabledConfig.serverCsrfTokenRepository.generateToken(any())
+        } returns Mono.just(this.token)
```

Comment:
In order to be consistent with our Java tests, we should continue using companion objects rather than getting the bean from the context. 
It seems with MockK we cannot create a mock in the companion object, but we can use the `mockkObject` to achieve the same behaviour.
```suggestion
        mockkObject(MultipartFormDataNotEnabledConfig.TOKEN_REPOSITORY)
        every {
            MultipartFormDataNotEnabledConfig.TOKEN_REPOSITORY.loadToken(any())
        } returns Mono.just(this.token)
        every {
            MultipartFormDataNotEnabledConfig.TOKEN_REPOSITORY.generateToken(any())
        } returns Mono.just(this.token)
```